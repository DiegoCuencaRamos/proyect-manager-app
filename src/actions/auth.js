import { auth, GoogleAuthProvider, FacebookAuthProvider, TwitterAuthProvider, GithubAuthProvider } from "../firebase"

// LOGIN
const login = (uid) => ({
    type: 'auth/login',
    payload: uid
})

// Login-middleware
const startGoogleLogin = () => {
    return (dispatch, getState) => {
        return auth.signInWithRedirect(GoogleAuthProvider)
            .catch(e => console.log(e.message))
    }
}

const startFacebookLogin = () => {
    return (dispatch, getState) => {
        return auth.signInWithRedirect(FacebookAuthProvider)
            .catch(e => console.log(e.message))
    }
}

const startTwitterLogin = () => {
    return (dispatch, getState) => {
        return auth.signInWithRedirect(TwitterAuthProvider)
            .catch(e => console.log(e.message))
    }
}

const startGithubLogin = () => {
    return (dispatch, getState) => {
        return auth.signInWithRedirect(GithubAuthProvider)
            .catch(e => console.log(e.message))
    }
}

const startEmailLogin = (email, password, setErrorMessage) => {
    return (dispath, getState) => {
        return auth.signInWithEmailAndPassword(email, password)
            .catch(e => {
                console.log(e.message)
                if(e.message === 'The password is invalid or the user does not have a password.') {
                    setErrorMessage('he password is invalid')
                } else {
                    setErrorMessage(e.message)
                }  
            })
    }
}

const startEmailSignUp = (email, password, setErrorMessage) => {
    return (dispatch, getState) => {
        return auth.createUserWithEmailAndPassword(email, password)
            .catch(e => {
                console.log(e.message)
                setErrorMessage(e.message)
            })
    }
}

// LOGOUT
const logout = () => ({
    type: 'auth/logout'
})

const startLogout = () => {
    return (dipatch, getState) => {
        return auth.signOut()
    }
}

export {
    login,
    startGoogleLogin,
    startFacebookLogin,
    startTwitterLogin,
    startGithubLogin,
    startEmailLogin,
    startEmailSignUp,
    logout,
    startLogout
}