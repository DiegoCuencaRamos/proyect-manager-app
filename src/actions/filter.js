const searchFilterChanged = (searchText = '') => ({
    type: 'filter/serachFilterChanged',
    payload: searchText
})

const sortByFilterChanged = (sortBy) => ({
    type: 'filter/sortByFilterChanged',
    payload: sortBy
})

const visualizeModeFilterChanged = (visualizeMode = 'list') => ({
    type: 'filter/visualizeModeFilterChanged',
    payload: visualizeMode
})

export { searchFilterChanged, sortByFilterChanged, visualizeModeFilterChanged }