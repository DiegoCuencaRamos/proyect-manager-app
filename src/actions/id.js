const setProyectId = id => ({
    type: 'id/setProyectId',
    payload: id
})

const setTaskId = id => ({
    type: 'id/setTaskId',
    payload: id
})

export { setProyectId, setTaskId }