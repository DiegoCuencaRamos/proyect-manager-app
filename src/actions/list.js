const setTextFilter = text => ({
    type: 'list/setTextFilter',
    payload: text
})

const setSortBy = sortBy => ({
    type: 'list/setSortBy',
    payload: sortBy
})

const setCurrentPage = currentPage => ({
    type: 'list/setCurrentPage',
    payload: currentPage
})

const setPageLimit = pageLimit => ({
    type: 'list/setPageLimit',
    payload: pageLimit
})

export {
    setTextFilter,
    setSortBy,
    setCurrentPage,
    setPageLimit
}