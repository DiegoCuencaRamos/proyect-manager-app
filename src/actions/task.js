import database from '../firebase'

// 1. Add task
const addTask = (task) => ({
    type: 'task/addTask',
    payload: task
})

const startAddTask = (taskData = {}) => {
    return (dispatch, getState) => {
        const {
            proyectId = '', 
            name = '', 
            description = '', 
            status = ''
        } = taskData
        const task = { proyectId, name, description, status }

        return database.ref('/tasks').push(task)
            .then(ref => {
                dispatch(addTask({
                    id: ref.key,
                    ...task
                }))
            })     
    }
}

// 2. Edit task
const editTask = (id, updates) => ({
    type: 'task/editTask',
    payload: {
        id,
        updates
    }
})

const startEditTask = (id, updates) => {
    return (dispatch, getState) => {

        return database.ref(`/tasks/${id}`).update(updates)
            .then(() => {
                dispatch(editTask(id, updates))
            })
    }
}

// 3. Remove task
const removeTask = (id) => ({
    type: 'task/removeTask',
    payload: id
})

const startRemoveTask = (id) => {
    return (dispatch, getState) => {

        return database.ref(`/tasks/${id}`).remove()
            .then(() => {
                dispatch(removeTask(id))
            })
    }
}

// 4. Set tasks
const setTasks = (tasks) => ({
    type: 'task/setTasks',
    payload: tasks
})

const startSetTasks = () => {
    return (dispatch, getState) => {

        return database.ref(`/tasks`).once('value')
            .then(dataSnapshot => {
                const tasks = []

                dataSnapshot.forEach(dataChildSnapshot => {
                    tasks.push({
                        id: dataChildSnapshot.key,
                        ...dataChildSnapshot.val()
                    })
                })

                dispatch(setTasks(tasks))
            })
    }
}

export { 
    addTask, 
    startAddTask,
    editTask, 
    startEditTask,
    removeTask,
    startRemoveTask,
    setTasks,
    startSetTasks
}