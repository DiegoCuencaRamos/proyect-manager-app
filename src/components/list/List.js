import React, { useContext, useEffect, useState } from 'react'
import { useSelector } from 'react-redux'
import DashboardContext from '../../contexts/dashboard-context'
import getVisibleProyects from '../../selectors/proyects'
import getVisibleTasks from '../../selectors/tasks'
import ListFilters from './ListFilters'
import ListHeader from './ListHeader'
import ListBody from './ListBody'
import Pagination from '../pagination/Pagination'


const List = () => {
    // Context
    const isProyect = useContext(DashboardContext)
    // Variables
    const items = useSelector(state => state[ isProyect ? 'proyects' : 'tasks' ]) 
    const filters = useSelector(state => state.filters)
    const proyectId = isProyect ? undefined : useSelector(state => state.ids.proyectId)
    const filteredItems = isProyect ? getVisibleProyects(items, filters) : getVisibleTasks(proyectId, items, filters)
    // State 
    const [ currentItems, setCurrentItems ] = useState([])
    const [ pageLimit, setPageLimit ] = useState(5) // Este puede ir en el calendar tambiÃ©n, estoy probando.

    useEffect(() => {
        setCurrentItems(filteredItems)
    }, [items])

    console.log('1: ', filteredItems)

    // Events
    const onItemsChange = (data) => {
        const { currentPage, pageLimit } = data
        const startItem = (currentPage - 1) * pageLimit
        const endItem = currentPage * pageLimit
        const currentItems = filteredItems.slice(startItem, endItem)

        console.log('2: ', filteredItems)

        setCurrentItems(currentItems)
    }

    const onPageLimitChange = (pageLimit) => {
        setPageLimit(pageLimit)
    }

    // Render
    return (
        <div className="list">
            <div className="container">
                <ListFilters />
                <div className="list__table">
                    <ListHeader />
                    <ListBody currentItems={currentItems} />  
                    <Pagination 
                        filteredItems={filteredItems}
                        totalItems={filteredItems.length} 
                        pageLimit={pageLimit}
                        onPageLimitChange={onPageLimitChange}
                        onItemsChange={onItemsChange}
                        pageNeighbours={1}
                    />                  
                </div>
            </div>
        </div>
    )
}

export default List