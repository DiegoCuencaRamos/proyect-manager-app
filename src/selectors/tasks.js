const getVisibleTasks = (proyectId, tasks, { searchText, sortBy }) => {
    return tasks
        // Get filtered tasks by proyect
        .filter(task => task.proyectId === proyectId)
        // Get filtered tasks by searchText
        .filter(task => task.name.toLowerCase().includes(searchText.toLowerCase()))
        // Order tasks by criteria
        .sort((a, b) => {
            if(sortBy === 'name') {
                return a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1
            }
        }) 
}

export default getVisibleTasks