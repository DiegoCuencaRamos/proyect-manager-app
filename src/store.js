import { createStore } from 'redux'
import rootReducer from './reducer'

const enhancer = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()

const store = createStore(
    rootReducer,
    /* preloadedState. */
    enhancer,
)

export default store