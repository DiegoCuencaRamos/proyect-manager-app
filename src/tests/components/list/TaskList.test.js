import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router'
import ListContextProvider from '../../../contexts/ListContext'
import { Provider } from 'react-redux'
import store from '../../../store'
import List from '../../../components/list/List'
import { tasks } from '../../fixures/tasks'
import { addTask } from '../../../actions/task'
import { setProyectId } from '../../../actions/id'


const renderWithWrapper = () => mount(
    <MemoryRouter initialEntries={[{ key: 'static' }]}>
        <ListContextProvider>
            <Provider store={store}>
                <List />
            </Provider>
        </ListContextProvider>
    </MemoryRouter>
)


// Rendering
test('Should render List correctly with no items', () => {
    const wrapper = renderWithWrapper()
    expect(wrapper).toMatchSnapshot()
})

test('Should render List correctly with tasks', () => {
    const proyectId = tasks[0].proyectId
    // Set tasks to the store
    tasks.forEach(task => {
        store.dispatch(addTask(task))
    })
    // Set proyectId to the store
    store.dispatch(setProyectId(proyectId))

    const wrapper = renderWithWrapper()
    expect(wrapper).toMatchSnapshot()
})

// Events
describe('Should handle Task List events:', () => {

    const id = tasks[0].id
    let wrapper

    beforeEach(() => {
        wrapper = renderWithWrapper()
    })
    
    test('Should navigate to Task Page on proyect clicked', () => {    
        wrapper.find(`.list__body ListItem#${id} Link`).simulate('click', { button: 0 })
        expect(wrapper.find('Router').prop('history').location.pathname).toBe(`/task/${id}`)
    })
    
    test('Should navigate to Edit Task Page on edit button clicked', () => {   
        // Simulate dropdown-button clicked 
        wrapper.find(`.list__body ListItem#${id} .list__dropdown-button`).simulate('click')
        // Simulate edit button clicked 
        wrapper.find(`.list__body ListItem#${id} .editBtn`).simulate('click', {
            target: {
                dataset: { id }
            }
        })
        expect(wrapper.find('Router').prop('history').location.pathname).toBe(`/edit-task/${id}`)
    })
})

