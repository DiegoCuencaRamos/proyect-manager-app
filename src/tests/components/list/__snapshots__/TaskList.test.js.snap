// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render List correctly with no items 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "static",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "static",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "static",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ListContextProvider>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <List>
          <section
            className="list"
          >
            <div
              className="container"
            >
              <ListFilters>
                <div
                  className="filters"
                >
                  <input
                    className="filters__item"
                    onChange={[Function]}
                    placeholder="Search"
                    type="text"
                    value=""
                  />
                </div>
              </ListFilters>
              <ListTable
                filteredItems={Array []}
              >
                <div
                  className="list__task-table"
                >
                  <ListHeader>
                    <div
                      className="list__header"
                    >
                      <div
                        className="list__header-item"
                      >
                        <div
                          className="list__name"
                          onClick={[Function]}
                        >
                          <span>
                            Name
                          </span>
                          <i
                            className="fas fa-sort-down"
                          />
                        </div>
                        <p
                          className="list__status"
                        >
                          Status
                        </p>
                        <p
                          className="list__dropdown"
                        >
                          Options
                        </p>
                      </div>
                    </div>
                  </ListHeader>
                  <ListBody
                    currentItems={Array []}
                  >
                    <div
                      className="list__body"
                    />
                  </ListBody>
                </div>
              </ListTable>
              <Pagination
                pageNeighbours={1}
                totalItems={0}
              >
                <div
                  className="pagination"
                >
                  <PaginationFilter>
                    <div
                      className="pagination__limit"
                    >
                      <select
                        className="filters__item"
                        onChange={[Function]}
                        value={5}
                      >
                        <option
                          value="2"
                        >
                          2 items
                        </option>
                        <option
                          value="5"
                        >
                          5 items
                        </option>
                        <option
                          value="10"
                        >
                          10 items
                        </option>
                        <option
                          value="20"
                        >
                          20 items
                        </option>
                      </select>
                    </div>
                  </PaginationFilter>
                  <PaginationPages
                    pageNeighbours={1}
                    pages={Array []}
                    totalPages={0}
                  >
                    <nav
                      className="pagination__pages"
                    >
                      <ul />
                    </nav>
                  </PaginationPages>
                </div>
              </Pagination>
            </div>
          </section>
        </List>
      </Provider>
    </ListContextProvider>
  </Router>
</MemoryRouter>
`;

exports[`Should render List correctly with tasks 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "static",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "static",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "static",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ListContextProvider>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <List>
          <section
            className="list"
          >
            <div
              className="container"
            >
              <ListFilters>
                <div
                  className="filters"
                >
                  <input
                    className="filters__item"
                    onChange={[Function]}
                    placeholder="Search"
                    type="text"
                    value=""
                  />
                </div>
              </ListFilters>
              <ListTable
                filteredItems={
                  Array [
                    Object {
                      "color": "green",
                      "description": "This is my task fixure 1",
                      "endDate": 1616108400000,
                      "id": "t1",
                      "name": "Task 1",
                      "proyectId": "p1",
                      "startDate": 1615762800000,
                      "status": "todo",
                    },
                    Object {
                      "color": "brown",
                      "description": "This is my task fixure 2",
                      "endDate": 1616540400000,
                      "id": "t2",
                      "name": "Task 2",
                      "proyectId": "p1",
                      "startDate": 1616194800000,
                      "status": "doing",
                    },
                    Object {
                      "color": "purple",
                      "description": "This is my task fixure 3",
                      "endDate": 1616968800000,
                      "id": "t3",
                      "name": "Task 3",
                      "proyectId": "p3",
                      "startDate": 1616626800000,
                      "status": "done",
                    },
                  ]
                }
              >
                <div
                  className="list__task-table"
                >
                  <ListHeader>
                    <div
                      className="list__header"
                    >
                      <div
                        className="list__header-item"
                      >
                        <div
                          className="list__name"
                          onClick={[Function]}
                        >
                          <span>
                            Name
                          </span>
                          <i
                            className="fas fa-sort-down"
                          />
                        </div>
                        <p
                          className="list__status"
                        >
                          Status
                        </p>
                        <p
                          className="list__dropdown"
                        >
                          Options
                        </p>
                      </div>
                    </div>
                  </ListHeader>
                  <ListBody
                    currentItems={
                      Array [
                        Object {
                          "color": "green",
                          "description": "This is my task fixure 1",
                          "endDate": 1616108400000,
                          "id": "t1",
                          "name": "Task 1",
                          "proyectId": "p1",
                          "startDate": 1615762800000,
                          "status": "todo",
                        },
                        Object {
                          "color": "brown",
                          "description": "This is my task fixure 2",
                          "endDate": 1616540400000,
                          "id": "t2",
                          "name": "Task 2",
                          "proyectId": "p1",
                          "startDate": 1616194800000,
                          "status": "doing",
                        },
                        Object {
                          "color": "purple",
                          "description": "This is my task fixure 3",
                          "endDate": 1616968800000,
                          "id": "t3",
                          "name": "Task 3",
                          "proyectId": "p3",
                          "startDate": 1616626800000,
                          "status": "done",
                        },
                      ]
                    }
                  >
                    <div
                      className="list__body"
                    >
                      <ListItem
                        color="green"
                        description="This is my task fixure 1"
                        endDate={1616108400000}
                        id="t1"
                        key="t1"
                        name="Task 1"
                        proyectId="p1"
                        startDate={1615762800000}
                        status="todo"
                      >
                        <div
                          className="list__body-item"
                          key="t1"
                        >
                          <Link
                            className="list__name"
                            data-id="t1"
                            to="/task/t1"
                          >
                            <LinkAnchor
                              className="list__name"
                              data-id="t1"
                              href="/task/t1"
                              navigate={[Function]}
                            >
                              <a
                                className="list__name"
                                data-id="t1"
                                href="/task/t1"
                                onClick={[Function]}
                              >
                                Task 1
                              </a>
                            </LinkAnchor>
                          </Link>
                          <p
                            className="list__status list__status--todo"
                          >
                            todo
                          </p>
                          <ListDropdown
                            id="t1"
                          >
                            <div
                              className="list__dropdown"
                            >
                              <ListDropdownButton
                                onButtonClicked={[Function]}
                              >
                                <div
                                  className="list__dropdown-button"
                                  onClick={[Function]}
                                >
                                  <i
                                    className="fas fa-sliders-h"
                                  />
                                  <i
                                    className="fas fa-sort-down"
                                  />
                                </div>
                              </ListDropdownButton>
                              <ListDropdownContent
                                id="t1"
                                onListClicked={[Function]}
                                open={false}
                              />
                            </div>
                          </ListDropdown>
                        </div>
                      </ListItem>
                      <ListItem
                        color="brown"
                        description="This is my task fixure 2"
                        endDate={1616540400000}
                        id="t2"
                        key="t2"
                        name="Task 2"
                        proyectId="p1"
                        startDate={1616194800000}
                        status="doing"
                      >
                        <div
                          className="list__body-item"
                          key="t2"
                        >
                          <Link
                            className="list__name"
                            data-id="t2"
                            to="/task/t2"
                          >
                            <LinkAnchor
                              className="list__name"
                              data-id="t2"
                              href="/task/t2"
                              navigate={[Function]}
                            >
                              <a
                                className="list__name"
                                data-id="t2"
                                href="/task/t2"
                                onClick={[Function]}
                              >
                                Task 2
                              </a>
                            </LinkAnchor>
                          </Link>
                          <p
                            className="list__status list__status--doing"
                          >
                            doing
                          </p>
                          <ListDropdown
                            id="t2"
                          >
                            <div
                              className="list__dropdown"
                            >
                              <ListDropdownButton
                                onButtonClicked={[Function]}
                              >
                                <div
                                  className="list__dropdown-button"
                                  onClick={[Function]}
                                >
                                  <i
                                    className="fas fa-sliders-h"
                                  />
                                  <i
                                    className="fas fa-sort-down"
                                  />
                                </div>
                              </ListDropdownButton>
                              <ListDropdownContent
                                id="t2"
                                onListClicked={[Function]}
                                open={false}
                              />
                            </div>
                          </ListDropdown>
                        </div>
                      </ListItem>
                      <ListItem
                        color="purple"
                        description="This is my task fixure 3"
                        endDate={1616968800000}
                        id="t3"
                        key="t3"
                        name="Task 3"
                        proyectId="p3"
                        startDate={1616626800000}
                        status="done"
                      >
                        <div
                          className="list__body-item"
                          key="t3"
                        >
                          <Link
                            className="list__name"
                            data-id="t3"
                            to="/task/t3"
                          >
                            <LinkAnchor
                              className="list__name"
                              data-id="t3"
                              href="/task/t3"
                              navigate={[Function]}
                            >
                              <a
                                className="list__name"
                                data-id="t3"
                                href="/task/t3"
                                onClick={[Function]}
                              >
                                Task 3
                              </a>
                            </LinkAnchor>
                          </Link>
                          <p
                            className="list__status list__status--done"
                          >
                            done
                          </p>
                          <ListDropdown
                            id="t3"
                          >
                            <div
                              className="list__dropdown"
                            >
                              <ListDropdownButton
                                onButtonClicked={[Function]}
                              >
                                <div
                                  className="list__dropdown-button"
                                  onClick={[Function]}
                                >
                                  <i
                                    className="fas fa-sliders-h"
                                  />
                                  <i
                                    className="fas fa-sort-down"
                                  />
                                </div>
                              </ListDropdownButton>
                              <ListDropdownContent
                                id="t3"
                                onListClicked={[Function]}
                                open={false}
                              />
                            </div>
                          </ListDropdown>
                        </div>
                      </ListItem>
                    </div>
                  </ListBody>
                </div>
              </ListTable>
              <Pagination
                pageNeighbours={1}
                totalItems={3}
              >
                <div
                  className="pagination"
                >
                  <PaginationFilter>
                    <div
                      className="pagination__limit"
                    >
                      <select
                        className="filters__item"
                        onChange={[Function]}
                        value={5}
                      >
                        <option
                          value="2"
                        >
                          2 items
                        </option>
                        <option
                          value="5"
                        >
                          5 items
                        </option>
                        <option
                          value="10"
                        >
                          10 items
                        </option>
                        <option
                          value="20"
                        >
                          20 items
                        </option>
                      </select>
                    </div>
                  </PaginationFilter>
                  <PaginationPages
                    pageNeighbours={1}
                    pages={
                      Array [
                        1,
                      ]
                    }
                    totalPages={1}
                  >
                    <nav
                      className="pagination__pages"
                    >
                      <ul>
                        <PaginationPageItem
                          key="0"
                          page={1}
                          pageNeighbours={1}
                          totalPages={1}
                        >
                          <li>
                            <a
                              className="pagination__active-page"
                              onClick={[Function]}
                            >
                              1
                            </a>
                          </li>
                        </PaginationPageItem>
                      </ul>
                    </nav>
                  </PaginationPages>
                </div>
              </Pagination>
            </div>
          </section>
        </List>
      </Provider>
    </ListContextProvider>
  </Router>
</MemoryRouter>
`;
